Examples/Framework/ML/src/NeuralCalibrator.cpp

```diff
@@ -102,81 +104,88 @@ void ActsExamples::NeuralCalibrator::calibrate(
+
+  assert(measurement.contains(Acts::eBoundLoc0) &&
+         "Measurement does not contain the required bound loc0");
+  assert(measurement.contains(Acts::eBoundLoc1) &&
+         "Measurement does not contain the required bound loc1");
+
+  auto boundLoc0 = measurement.subspace().indexOf(Acts::eBoundLoc0);
+  auto boundLoc1 = measurement.subspace().indexOf(Acts::eBoundLoc1);
+
+  Acts::Vector2 localPosition{measurement.effectiveParameters()[boundLoc0],
+                              measurement.effectiveParameters()[boundLoc1]};
+  Acts::Vector2 localCovariance{
+      measurement.effectiveCovariance()(boundLoc0, boundLoc0),
+      measurement.effectiveCovariance()(boundLoc1, boundLoc1)};
+
+  Acts::Vector3 dir = Acts::makeDirectionFromPhiTheta(
+      trackParameters[Acts::eBoundPhi], trackParameters[Acts::eBoundTheta]);
+  Acts::Vector3 globalPosition =
+      referenceSurface.localToGlobal(gctx, localPosition, dir);
+
+  // Rotation matrix. When applied to global coordinates, they
+  // are rotated into the local reference frame of the
+  // surface. Note that this such a rotation can be found by
+  // inverting a matrix whose columns correspond to the
+  // coordinate axes of the local coordinate system.
+  Acts::RotationMatrix3 rot =
+      referenceSurface.referenceFrame(gctx, globalPosition, dir).inverse();
+  std::pair<double, double> angles =
+      Acts::VectorHelpers::incidentAngles(dir, rot);
+
+  input[iInput++] = angles.first;
+  input[iInput++] = angles.second;
+  input[iInput++] = localPosition[0];
+  input[iInput++] = localPosition[1];
+  input[iInput++] = localCovariance[0];
+  input[iInput++] = localCovariance[1];
+  if (iInput != m_nInputs) {
+    throw std::runtime_error("Expected input size of " +
+                             std::to_string(m_nInputs) +
+                             ", got: " + std::to_string(iInput));
+  }
+
+  // Input is a single row, hence .front()
+  std::vector<float> output = m_model.runONNXInference(inputBatch).front();
+  // Assuming 2-D measurements, the expected params structure is:
+  // [           0,    nComponent[ --> priors
+  // [  nComponent,  3*nComponent[ --> means
+  // [3*nComponent,  5*nComponent[ --> variances
+  std::size_t nParams = 5 * m_nComponents;
+  if (output.size() != nParams) {
+    throw std::runtime_error("Got output vector of size " +
+                             std::to_string(output.size()) +
+                             ", expected size " + std::to_string(nParams));
+  }
 
-  std::visit(
-      [&](const auto& measurement) {
-        auto E = measurement.expander();
-        auto P = measurement.projector();
-        Acts::ActsVector<Acts::eBoundSize> fpar = E * measurement.parameters();
-        Acts::ActsSquareMatrix<Acts::eBoundSize> fcov =
-            E * measurement.covariance() * E.transpose();
-
-        Acts::Vector3 dir = Acts::makeDirectionFromPhiTheta(
-            fpar[Acts::eBoundPhi], fpar[Acts::eBoundTheta]);
```

Comment:
@gagnonlg changed to use track params here otherwise these will be 0

Reply:
That looks right to me yes -- seems like I hadn't tested this thoroughly enough... Thanks